templates/api.hbs

```diff
@@ -1,17 +1,39 @@
 {{>header}}
 
-import { Express } from 'express'
+import { Express, Request, Response } from 'express'
 import passport from 'passport'
+import multer from 'multer'
 import * as t from './types'
 import * as v from '../../validation'
+{{#if (hasOperationSupportingMultipart operations)}}
+import * as f from '../../impl/helpers/{{lowerCase name}}MultipartHelper'
+{{/if}}
 import { Api } from '../../models'
 
 export default function(app: Express, impl: t.{{className name}}Api) {
 	{{#each operations}}
 	app.{{lowerCase httpMethod}}(
 		{{{stringLiteral (pathTemplate fullPath)}}},
 		{{>frag/apiSecurityRequirements}}
-		function (req, res) {
+		{{#if (hasOperationSupportingMultipart .)}}
+		{{#with (getFileProperties requestBody.defaultContent.schema.properties)}}
+		(req, res, next) => {
+			f.upload.fields([{{#each this}}{ name: {{{stringLiteral name}}}, maxCount: {{#if maxCount}}{{{maxCount}}}{{else}}1{{/if}} }{{#hasMore}}, {{/hasMore}}{{/each}}])(req, res, (error) => {
+				if (error?.code === 'LIMIT_UNEXPECTED_FILE') {
+					res.status(400).send(`You have exceeded the maximum number of files allowed for the field '${error.field}'`)
+					return
+				} else {
+					next(error)
+				}
+			})
+		},
+		{{~#each this}}
+		{{#if minCount}}v.verifyMinimumFiles({{{stringLiteral name}}}, {{{minCount}}}),{{/if}}
```

Comment:
I think the `{{#if}}` should be on its own line, and the `{{/if}}` so that if it's false we don't get a blank line.

Handlebars strips lines if they _only_ contain a block command, ie. `{{#...}`